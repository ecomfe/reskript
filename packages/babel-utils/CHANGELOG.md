# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.2.2](https://github.com/ecomfe/reskript/compare/v2.2.1...v2.2.2) (2021-09-12)

**Note:** Version bump only for package @reskript/babel-utils





## [2.2.1](https://github.com/ecomfe/reskript/compare/v2.2.0...v2.2.1) (2021-09-12)

**Note:** Version bump only for package @reskript/babel-utils





# [2.2.0](https://github.com/ecomfe/reskript/compare/v2.1.0...v2.2.0) (2021-09-12)

**Note:** Version bump only for package @reskript/babel-utils





# [2.1.0](https://github.com/ecomfe/reskript/compare/v2.0.0...v2.1.0) (2021-08-31)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0](https://github.com/ecomfe/reskript/compare/v2.0.0-beta.5...v2.0.0) (2021-08-26)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0-beta.5](https://github.com/ecomfe/reskript/compare/v2.0.0-beta.4...v2.0.0-beta.5) (2021-08-26)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0-beta.4](https://github.com/ecomfe/reskript/compare/v2.0.0-beta.3...v2.0.0-beta.4) (2021-08-25)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0-beta.3](https://github.com/ecomfe/reskript/compare/v2.0.0-beta.2...v2.0.0-beta.3) (2021-08-25)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0-beta.2](https://github.com/ecomfe/reskript/compare/v2.0.0-beta.1...v2.0.0-beta.2) (2021-08-25)

**Note:** Version bump only for package @reskript/babel-utils





# [2.0.0-beta.1](https://github.com/ecomfe/reskript/compare/v1.15.2...v2.0.0-beta.1) (2021-08-25)

**Note:** Version bump only for package @reskript/babel-utils





## [1.15.2](https://github.com/ecomfe/reskript/compare/v1.15.1...v1.15.2) (2021-08-24)

**Note:** Version bump only for package @reskript/babel-utils





## [1.15.1](https://github.com/ecomfe/reskript/compare/v1.15.0...v1.15.1) (2021-08-20)

**Note:** Version bump only for package @reskript/babel-utils





# [1.15.0](https://github.com/ecomfe/reskript/compare/v1.14.2...v1.15.0) (2021-08-19)

**Note:** Version bump only for package @reskript/babel-utils





## [1.14.2](https://github.com/ecomfe/reskript/compare/v1.14.1...v1.14.2) (2021-08-17)

**Note:** Version bump only for package @reskript/babel-utils





## [1.14.1](https://github.com/ecomfe/reskript/compare/v1.14.0...v1.14.1) (2021-08-13)

**Note:** Version bump only for package @reskript/babel-utils





# [1.14.0](https://github.com/ecomfe/reskript/compare/v1.13.1...v1.14.0) (2021-08-12)

**Note:** Version bump only for package @reskript/babel-utils





## [1.13.1](https://github.com/ecomfe/reskript/compare/v1.13.0...v1.13.1) (2021-08-07)

**Note:** Version bump only for package @reskript/babel-utils





# [1.13.0](https://github.com/ecomfe/reskript/compare/v1.12.2...v1.13.0) (2021-08-05)

**Note:** Version bump only for package @reskript/babel-utils





## [1.12.2](https://github.com/ecomfe/reskript/compare/v1.12.1...v1.12.2) (2021-08-02)

**Note:** Version bump only for package @reskript/babel-utils





## [1.12.1](https://github.com/ecomfe/reskript/compare/v1.12.0...v1.12.1) (2021-08-02)


### Bug Fixes

* **babel:** babel-utils少了个依赖 ([477a159](https://github.com/ecomfe/reskript/commit/477a159f763261533d91c0acc813d42f012289ac))





# [1.12.0](https://github.com/ecomfe/reskript/compare/v1.11.2...v1.12.0) (2021-07-29)

**Note:** Version bump only for package @reskript/babel-utils





## [1.11.2](https://github.com/ecomfe/reskript/compare/v1.11.1...v1.11.2) (2021-07-28)

**Note:** Version bump only for package @reskript/babel-utils





## [1.11.1](https://github.com/ecomfe/reskript/compare/v1.11.0...v1.11.1) (2021-07-25)

**Note:** Version bump only for package @reskript/babel-utils





# [1.11.0](https://github.com/ecomfe/reskript/compare/v1.10.3...v1.11.0) (2021-07-23)


### Performance Improvements

* **babel:** 优化对import语句的处理的性能 ([bc8e0dc](https://github.com/ecomfe/reskript/commit/bc8e0dc3c7fcee555a664b5168cc30683e421f19))





## [1.10.3](https://github.com/ecomfe/reskript/compare/v1.10.2...v1.10.3) (2021-07-22)


### Bug Fixes

* **babel:** 对组件函数的检测使用更严格的模式 ([#97](https://github.com/ecomfe/reskript/issues/97)) ([61fc2df](https://github.com/ecomfe/reskript/commit/61fc2df685eb3ad628592cbc0c0e1252def4de7d))





## [1.10.2](https://github.com/ecomfe/reskript/compare/v1.10.1...v1.10.2) (2021-07-22)


### Bug Fixes

* **babel:** 调整babel插件的顺序避免代码转换出错 ([#97](https://github.com/ecomfe/reskript/issues/97)) ([f9d6f97](https://github.com/ecomfe/reskript/commit/f9d6f979da88fe75f0029b519e416b090cb85f73))





## [1.10.1](https://github.com/ecomfe/reskript/compare/v1.10.0...v1.10.1) (2021-07-20)

**Note:** Version bump only for package @reskript/babel-utils





# [1.10.0](https://github.com/ecomfe/reskript/compare/v1.9.0...v1.10.0) (2021-07-20)


### Features

* **dev:** 增加一个babel插件为React组件注入对应源码路径 ([#91](https://github.com/ecomfe/reskript/issues/91)) ([fb0132d](https://github.com/ecomfe/reskript/commit/fb0132d31dc83128b8373da7b38ca0c3d32b4a9c))
