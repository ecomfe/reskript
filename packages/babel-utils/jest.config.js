module.exports = {
    preset: 'ts-jest/presets/default',
    testMatch: ['**/__tests__/**/*.test.ts'],
    transformIgnorePatterns: [
        '\\/dist\\/',
        'node_modules',
    ],
};
