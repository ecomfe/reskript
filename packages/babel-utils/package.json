{
  "name": "@reskript/babel-utils",
  "version": "2.2.2",
  "main": "dist/index.js",
  "license": "MIT",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rm -rf dist",
    "build": "tsc -p tsconfig.build.json",
    "test": "jest",
    "lint": "eslint --max-warnings=0 src"
  },
  "devDependencies": {
    "@babel/core": "^7.15.5",
    "@babel/traverse": "^7.15.4",
    "@types/babel__core": "^7.1.16",
    "@types/jest": "^27.0.1",
    "@types/node": "^16.9.1",
    "eslint": "^7.32.0",
    "jest": "^27.1.1",
    "ts-jest": "^27.0.5",
    "typescript": "^4.4.3"
  },
  "peerDependencies": {
    "@babel/core": "^7.14.6",
    "@babel/traverse": "^7.14.7"
  },
  "dependencies": {
    "glob": "^7.1.7"
  }
}
