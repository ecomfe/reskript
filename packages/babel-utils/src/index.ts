export * from './importReact';
export * from './testRunner';
export * from './component';
export * from './import';
export * from './traverse';
