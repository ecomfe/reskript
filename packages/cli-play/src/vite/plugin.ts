import fs from 'node:fs/promises';
import {Plugin} from 'vite';
import {buildEntryScript, resolveEntryPath, BuildEntryOptions} from '../utils/entry.js';

interface Options extends BuildEntryOptions {
    enableConcurrentMode: boolean;
}

export default function playEntryPlugin(options: Options): Plugin {
    return {
        name: 'reskript:play-entry',
        enforce: 'pre',
        resolveId(id) {
            if (id === '/playground-entry.jsx') {
                return id;
            }
        },
        async load(id) {
            if (id === '/playground-entry.jsx') {
                const content = await fs.readFile(resolveEntryPath(options.enableConcurrentMode), 'utf-8');
                return buildEntryScript(content, options);
            }
        },
    };
}
