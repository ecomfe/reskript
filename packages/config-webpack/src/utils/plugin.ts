import {Compilation, Compiler} from 'webpack';
import HtmlWebpackPlugin from 'html-webpack-plugin';

export class TransformHtmlWebpackPlugin {
    readonly transform: (html: string) => string;

    constructor(transform: (html: string) => string) {
        this.transform = transform;
    }

    apply(compiler: Compiler) {
        compiler.hooks.compilation.tap('transform-html-webpack-plugin', compilation => this.inject(compilation));
    }

    inject(compilation: Compilation) {
        const {afterTemplateExecution} = HtmlWebpackPlugin.getHooks(compilation);
        afterTemplateExecution.tap(
            'transform-html-webpack-plugin',
            data => ({...data, html: this.transform(data.html)})
        );
    }
}
