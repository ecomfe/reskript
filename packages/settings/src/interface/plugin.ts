import {CommandInput} from './command.js';
import {ProjectSettings} from './project.js';

type SettingsPluginReturn = ProjectSettings | Promise<ProjectSettings>;

export type SettingsPluginItem = (current: ProjectSettings, cmd: CommandInput) => SettingsPluginReturn;

export type MayBeSettingsPlugin = SettingsPluginItem | null | undefined | false;

export type SettingsPlugin = MayBeSettingsPlugin | MayBeSettingsPlugin[];
