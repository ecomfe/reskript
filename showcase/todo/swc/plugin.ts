import path from 'path';
import {fileURLToPath} from 'url';
import {SettingsPluginItem} from '@reskript/settings';

const currentDirectory = path.dirname(fileURLToPath(import.meta.url));

export default (): SettingsPluginItem => {
    return settings => {
        if (settings.driver !== 'webpack') {
            throw new Error('VITE IS BAD');
        }

        return {
            ...settings,
            build: {
                ...settings.build,
                finalize: async (configIn, buildEntry, internals) => {
                    const loadingRules = [
                        internals.rules.less(buildEntry),
                        internals.rules.css(buildEntry),
                        internals.rules.svg(buildEntry),
                        internals.rules.file(buildEntry),
                    ] as const;
                    const builtinRules = await Promise.all(loadingRules);
                    const config = await settings.build.finalize(configIn, buildEntry, internals);
                    config.module.rules = [
                        {
                            test: /\.[jt]sx?$/,
                            include: path.join(currentDirectory, '..', 'src'),
                            use: [
                                {
                                    loader: path.join(currentDirectory, 'loader'),
                                    options: {
                                        externalHelpers: true,
                                        parser: {
                                            syntax: 'typescript',
                                            dynamicImport: true,
                                            decorators: false,
                                            jsx: true,
                                            importAssertions: true,
                                        },
                                        experimental: {
                                            keepImportAssertions: true,
                                            plugins: [],
                                        },
                                        transform: {
                                            legacyDecorator: false,
                                            decoratorMetadata: false,
                                            useDefineForClassFields: false,
                                            react: {
                                                importSource: '@emotion/react',
                                                runtime: 'automatic',
                                                throwIfNamespace: true,
                                                development: false,
                                                useBuiltins: true,
                                                refresh: false,
                                            },
                                            optimizer: {
                                                simplify: false,
                                                globals: {
                                                    envs: {
                                                        NODE_ENV: '"production"',
                                                        'skr.features.batch': 'true',
                                                    },
                                                },
                                            },
                                        },
                                    },
                                },
                            ],
                        },
                        ...builtinRules,
                    ];
                    return config;
                },
            },
        };
    };
};
