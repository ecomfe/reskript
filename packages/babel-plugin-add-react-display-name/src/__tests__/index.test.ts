import {createTestRunner} from '@reskript/babel-utils';
import plugin from '../index';

const runSuite = createTestRunner(
    __dirname,
    {test, expect},
    {
        presets: [
            '@babel/preset-react',
        ],
        plugins: [plugin],
    }
);

runSuite();
