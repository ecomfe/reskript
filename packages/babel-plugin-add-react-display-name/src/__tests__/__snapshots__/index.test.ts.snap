// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export default a render function 1`] = `
"// export default a render function
export default function () {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
}"
`;

exports[`export default function containing nested function component 1`] = `
"// export default function containing nested function component
export default function () {
  function Foo() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

  Foo.displayName = \\"Foo\\";
  return Foo;
}
;"
`;

exports[`export default named function 1`] = `
"// export default named function
export default function Foo() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
}
Foo.displayName = \\"Foo\\";"
`;

exports[`render function returns jsx 1`] = `
"// render function returns jsx
function renderFoo() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
}"
`;

exports[`return jsx 1`] = `
"// return jsx
function Foo() {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
}

Foo.displayName = \\"Foo\\";"
`;
